unittests/test_loader.py

```diff
@@ -58,3 +58,11 @@ def test_conflicted_checks(self):
     def test_load_error(self):
         self.assertRaises(OSError, self.loader.load_from_file,
                           'unittests/resources/checks/foo.py')
+
+    def test_load_invalid_sytax(self):
+        invalid_check = ('unittests/resources/checks_unlisted/'
+                         'invalid_syntax_check.py')
+        with self.assertRaises(SyntaxError) as e:
+            self.loader.load_from_file(invalid_check)
+
+        self.assertEqual(e.exception.filename, invalid_check)
```

Comment:
I know that `e` is visible outside the `with` statement, but does it make sense to have this assertion outside the `with` block?

Reply:
If you put the assertion inside the with block, it is not reachable since the with block exits once the exception is thrown.